stl/inc/algorithm

```diff
@@ -1774,6 +1898,107 @@ _NODISCARD _CONSTEXPR20 _FwdIt partition_point(_FwdIt _First, _FwdIt _Last, _Pr
     return _First;
 }
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::partition_point
+    class _Partition_point_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <forward_iterator _It, sentinel_for<_It> _Se, class _Pj = identity,
+            indirect_unary_predicate<projected<_It, _Pj>> _Pr>
+        _NODISCARD constexpr _It operator()(_It _First, _Se _Last, _Pr _Pred, _Pj _Proj = {}) const {
+            _Adl_verify_range(_First, _Last);
+            if constexpr (sized_sentinel_for<_Se, _It>) {
+                auto _UFirst       = _Get_unwrapped(_STD move(_First));
+                const auto _Length = _Get_unwrapped(_STD move(_Last)) - _UFirst;
+                _UFirst = _Partition_point_with_length(_STD move(_UFirst), _Length, _Pass_fn(_Pred), _Pass_fn(_Proj));
+                _Seek_wrapped(_First, _STD move(_UFirst));
+                return _First;

```

Comment:
Could that be `return _Rewrap_iterator(_First, _STD move(_UFirst));`

Reply:
`_Rewrap_iterator` takes a range `r` and unwrapped iterator `i`, and returns the result of rewrapping `i` into the result of `ranges::begin(r)`.

`_Rewrap_subrange(X, s)` takes either an unwrapped iterator or range `X` and an unwrapped `subrange` instance `s` and returns a wrapped subrange. 

There's no iterator overload of `_Rewrap_iterator` since we almost always already have an lvalue iterator laying around that we can simply `_Seek_wrapped`.